### PR TITLE
Bump `escrow`

### LINKED ISSUE #441: Escrow: outdated comment in harness.rs
### Motivation

The escrow has a comment stating that an [array should be changed to a vector](https://github.com/FuelLabs/sway-applications/blob/0132a4a89bb0f9c1a5d016db609adead87c43626/escrow/project/escrow-contract/tests/harness.rs#L4).

The contract seems to use vectors so that comment is likely invalid. Remove it.

### LINKED ISSUE #443: Escrow: harness.rs mentions unfinished tests
### Motivation

Block manipulation has been mentioned in the file. Investigate and amend any tests.

### LINKED ISSUE #654: Bump `escrow`
To:

- `forc 0.42.1`
- `fuel-core 0.18.3`
- `fuels-rs 0.42.0`

### PR DESCRIPTION
**Pull request by K1-R1**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)
- Documentation

## Changes

The following changes have been made:

- Bump application to `forc 0.42.1` and `fuel-core 0.18.3`
- Bump test harness to `fuels 0.42.0`

## Related Issues

Closes #654
Closes #441 
Closes #443

### CODE REVIEW

File: escrow/project/contracts/escrow-contract/tests/functions/core/take_payment.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ mod success {
         deposit(defaults.asset_amount, &defaults.asset_id, &buyer, 0).await;
 
         provider
-            .produce_blocks(origin_block + defaults.deadline, None)
+            .produce_blocks((origin_block as u64) + defaults.deadline, None)
```
Comment (Braqzen) [inline escrow/project/contracts/escrow-contract/tests/functions/core/take_payment.rs:43]: Are these casts type safe? I assume it's probably okay if clippy makes CI pass

File: escrow/project/contracts/escrow-contract/tests/utils/setup.rs [hunk 2]
```diff
@@ -15,13 +15,13 @@ abigen!(
     ),
     Contract(
         name = "MyAsset",
-        abi = "./contracts/escrow-contract/tests/artifacts/asset/out/debug/asset-abi.json"
+        abi = "./contracts/test-artifacts/asset/out/debug/asset-abi.json"
```
Comment (Braqzen) [inline escrow/project/contracts/escrow-contract/tests/utils/setup.rs:18]: As mentioned in another PR, this should be moved back into tests or updated to use the SDK coin/asset generation API

Comment (K1-R1) [reply escrow/project/contracts/escrow-contract/tests/utils/setup.rs:18]: This is required due to changes with nested projects and forc workspaces. An error is produced if the test artifacts are kept in their old places.

Comment (K1-R1) [reply escrow/project/contracts/escrow-contract/tests/utils/setup.rs:18]: Have re-written to use SDK

File: escrow/project/contracts/escrow-contract/tests/functions/core/resolve_dispute.rs [hunk 3]
```diff
@@ -97,8 +104,10 @@ mod success {
         .await;
         deposit(defaults.asset_amount, &defaults.asset_id, &buyer, 0).await;
 
-        assert_eq!(0, asset_amount(&defaults.asset_id, &buyer).await);
-        assert_eq!(0, asset_amount(&defaults.asset_id, &seller).await);
+        assert_eq!(
```
Comment (Braqzen) [inline escrow/project/contracts/escrow-contract/tests/functions/core/resolve_dispute.rs:107]: Buyer assertion seems to have been removed

**Review by Braqzen — COMMENTED** [review]
Same as the DAO, remove `rand` and use `AssetId::from()` to create a static value

**Review by Braqzen — APPROVED** [review]

**Pull request merged**